### PR TITLE
Functional update creation with position

### PR DESCRIPTION
**Pull request by thefloweringash**

### CODE REVIEW

**Comment by thefloweringash** [comment]
@chrisandreae  This is the direction I think I want to go with this code. There's currently a tension between the JSON schema and nicer error messages, with some overlap which I will eliminate. Have a look, let me know what you think.

File: test/unit/active_record_view_model/has_many_test.rb [hunk 1]
```diff
@@ -567,13 +683,13 @@ def test_functional_update_update_success
   end
 
   def test_functional_update_update_failure
-    cnew                = Child.create(parent: Parent.create).id
```
Comment (thefloweringash) [inline test/unit/active_record_view_model/has_many_test.rb]: stray whitespace

File: test/unit/active_record_view_model/has_many_test.rb [hunk 2]
```diff
@@ -511,12 +511,128 @@ def test_functional_update_append
     result          = ParentView.deserialize_from_view(append_view)
     @parent1.reload
 
-    created_children = result.children[-2,2].map(&:id)
```
Comment (thefloweringash) [inline test/unit/active_record_view_model/has_many_test.rb]: stray whitespace

File: lib/active_record_view_model/update_data.rb [hunk 3]
```diff
@@ -32,53 +56,110 @@ class UpdateData
...
+          'additionalProperties' => false,
+          'required'             => [TYPE_ATTRIBUTE, VALUES_ATTRIBUTE]
         }
 
+      append = base_functional_update_schema.deep_merge(
+        {
+          'description' => 'collection append',
+          'properties'  => {
+            TYPE_ATTRIBUTE   => { 'enum' => [FunctionalUpdate::Type::Append.name.downcase] },
+            VALUES_ATTRIBUTE => { 'items' => viewmodel_update },
```
Comment (chrisandreae) [inline lib/active_record_view_model/update_data.rb]: can fold this into base

File: lib/active_record_view_model/update_data.rb [hunk 4]
```diff
@@ -32,53 +56,110 @@ class UpdateData
...
-            ACTIONS_ATTRIBUTE => { 'type'  => 'array',
-                                   'items' => collection_update_action }
+            ACTIONS_ATTRIBUTE => { 'type'  => 'array' }
+            # The ACTIONS_ATTRIBUTE could be accurately expressed as
+            #
+            #   { 'oneOf' => [append, update, remove] }
+            #
+            # but this produces completely unusable error messages.  Instead we
+            # specify it must be an array, and defer checking to the code that
+            # can determine the schema by inspecting the type field.
```
Comment (chrisandreae) [inline lib/active_record_view_model/update_data.rb]: In that case could you at least check here that it's a collection update base (allowing additional properties), and like you say do the subtype checking individually after match?

**Comment by chrisandreae** [comment]
lgtm

**Pull request merged**